src/libcore/slice/mod.rs

```diff
@@ -1575,6 +1575,134 @@ impl<T> [T] {
         sort::quicksort(self, |a, b| f(a).lt(&f(b)));
     }
 
+    /// Rearranges the elements of the slice such that the following properties hold:
+    ///
+    /// - The value at position `index` will be the same as if the array were sorted.
+    /// - Any value at position `i < index` will be less than or equal to any value at a position
+    ///   `j > index`.
+    ///
+    /// As an example, `v.sort_at_index(v.len() / 2)` can be used to find the median of the array.
+    /// This reordering is unstable (i.e. any number of equal elements may end up at position
+    /// `index`), in-place (i.e. does not allocate), and `O(n)` worst-case. This function is also
+    /// known as "kth element" in other libraries. Returns the new value at the given index.
```

Comment:
> Returns the new value at the given index.

We decided against this, right? The function returns nothing.

Reply:
Yes. Removed.